compiler/src/dotty/tools/dotc/transform/TreeChecker.scala

```diff
@@ -145,7 +145,7 @@ class TreeChecker extends Phase with SymTransformer {
     override def checkDerivedValueClass(clazz: Symbol, stats: List[Tree])(implicit ctx: Context) = ()
 
     def withDefinedSym[T](tree: untpd.Tree)(op: => T)(implicit ctx: Context): T = tree match {
-      case tree: DefTree =>
+      case tree: DefTree @unchecked =>
```

Comment:
Yeah looking at it, it should be an `untpd.DefTree`. Same thing for `case t: MemberDef` below.